File: fast_sync_dart_client/fast_sync_floor_dao/lib/src/metadata/sync_metadata_model.dart
import 'package:fast_sync_client/fast_sync_client.dart';
import 'package:json_annotation/json_annotation.dart';
part 'sync_metadata_model.g.dart';

@JsonSerializable()
class SyncMetadataModel extends SyncMetadata {
  final String metadataTableName = "SyncMetadata";
  const SyncMetadataModel(
      {required String id,
      required String syncZone,
      required String type,
      required num version,
      required num timestamp,
      required SyncOperationEnum syncOperation})
      : super(
            id: id,
            syncZone: syncZone,
            type: type,
            version: version,
            timestamp: timestamp,
            syncOperation: syncOperation);

  factory SyncMetadataModel.fromJson(Map<String, dynamic> json) =>
      _$SyncMetadataModelFromJson(json);

  Map<String, dynamic> toJson() => _$SyncMetadataModelToJson(this);

  static String createSchema() {
    return ('''
      CREATE TABLE IF NOT EXISTS ${Constants.metadataTableName} (
        id TEXT PRIMARY KEY,
        syncZone TEXT,
        type TEXT,
        version REAL,
        timestamp REAL,
        syncOperation INTEGER
      )
    ''');
  }
}

File: fast_sync_typescript_service_sdk/packages/fast-sync-mongoose-dao/src/mongoose-dao/metadata/sync_metadata_model.ts
import * as mongoose from 'mongoose';

export  const SyncMetadataSchema = new mongoose.Schema({
    id: {
        type: mongoose.Schema.Types.String,
        required: true,
        auto: true,
    },
    syncZone: {
        type: String,
        required: true,
    }, 
    type: {
        type: String,
        required: true,
    },
    version: {
        type: Number,
        required: true,
    },
    timestamp: {
        type: Number,
        required: true,
    },
  });
  
